ibm/service/db2/data_source_ibm_db2_connection_info.go

```diff
@@ -0,0 +1,221 @@
+// Copyright IBM Corp. 2024 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+/*
+ * IBM OpenAPI Terraform Generator Version: 3.96.0-d6dec9d7-20241008-212902
+ */
+
+package db2
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM/cloud-db2-go-sdk/db2saasv1"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM/go-sdk-core/v5/core"
+)
+
+func DataSourceIbmDb2ConnectionInfo() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIbmDb2ConnectionInfoRead,
+
+		Schema: map[string]*schema.Schema{
+			"deployment_id": &schema.Schema{
```

Comment:
why we are following different naming convention for same argument 
```
deployment_id
```
Above datasource has
```
x_db_profile
```

But description remains same "Encoded CRN deployment id."

Reply:
In datasource_connection_info, deployment_id which has to be the encoded CRN which is passed as path param.
Whereas autoscale APIs expect x_db_profile which can be passed as encoded CRN(but it also takes normal CRN and APIs work fine), and has to be passed as header.
so as per API definition, we added the code.

that's why, to avoid confusion, we added two different naming convention.

Reply:
As per https://developer.hashicorp.com/terraform/plugin/best-practices/naming we except the names to be nouns which will be user friendly and self explanatory . If you use generator tool you have capability to override the names of the attributes/arguments. 
That naming convention may be internal but for an end user they should some simple names which are self explanatory

Reply:
@hkantare , 
we can't override the name of x_db_profile -- it's passed as header.

Reply:
You can do that mapping in your datasource file

Reply:
getDb2SaasAutoscaleOptions.SetXDbProfile(d.Get("deployment_id").(string))

Reply:
Changed it to deployment_id @hkantare